src/ucp/core/ucp_request.c

```diff
@@ -194,7 +194,7 @@ UCS_PROFILE_FUNC_VOID(ucp_request_cancel, (worker, request),
         return;
     }
 
-    if (req->flags & UCP_REQUEST_FLAG_EXPECTED) {
+    if (req->flags & UCP_REQUEST_FLAG_RECV_TAG) {
```

Comment:
i think the real bug here is that UCP_REQUEST_FLAG_EXPECTED flag is not cleared after we remove it from expected queue

Reply:
discussed offline - EXPECTED flag is not really needed and can be removed, since tag_exp_remove does not require the request to be on exp queue